LICENSE

```diff
@@ -225,3 +225,4 @@ The text of each license is the standard Apache 2.0 license.
    mvnw files from https://github.com/apache/maven-wrapper Apache 2.0
    svg files from skywalking-ui/src/assets/icons: https://github.com/google/material-design-icons Apache 2.0
    ZipkinQueryHandler.java reference from zipkin2.server.internal.ZipkinQueryApiV2: https://github.com/openzipkin/zipkin  Apache 2.0
+   async-profiler parser files in server-library/library-async-profiler-jfr-parser from https://github.com/open-telemetry/opentelemetry-proto/tree/v0.7.0 Apache 2.0
```

Comment:
Can’t we just use their Java SDK instead of copying everything over here?

https://mvnrepository.com/artifact/tools.profiler/async-profiler/3.0

Reply:
I also found that not all files in `server-library/library-async-profiler-jfr-parser` are copied, e.g. oap-server/server-library/library-async-profiler-jfr-parser/src/main/java/org/apache/skywalking/oap/server/library/jfr/parser/convert/FrameTree.java

Reply:
The library you mentioned is used for javagent, and the main function of oap server is to parse the JFR file generated by the agent. It also has a corresponding [package](https://mvnrepository.com/artifact/tools.profiler/async-profiler-converter/3.0), but it is not easy to use. For example, some of its classes do not have package names, which makes it impossible to use it normally in java.

Reply:
Take a look at https://github.com/jvm-profiling-tools/ap-loader it was [mentioned by the author of async-profiler](https://github.com/async-profiler/async-profiler/issues/93#issuecomment-1328155966) and it fixes the issue you mentioned, every class has a package name in it 


```shell
tar -tf /Users/kezhenxu94/Downloads/ap-loader-all-3.0-9.jar | grep .class | pbcopy 
one/proto/Proto.class
one/converter/jfr2pprof$Method.class
one/converter/jfr2nflx$1.class
one/converter/jfr2pprof.class
one/converter/Classifier.class
one/converter/Arguments.class
one/converter/FlameGraph$Frame.class
one/converter/CollapsedStacks.class
one/converter/jfr2flame.class
one/converter/FlameGraph.class
one/converter/jfr2nflx.class
one/converter/Main.class
one/converter/jfr2pprof$Location.class
one/converter/jfr2flame$1.class
one/jfr/JfrField.class
one/jfr/MethodRef.class
one/jfr/StackTrace.class
one/jfr/JfrReader.class
one/jfr/Element.class
one/jfr/Dictionary$Visitor.class
one/jfr/ClassRef.class
one/jfr/Dictionary.class
one/jfr/JfrClass.class
one/jfr/event/GCHeapSummary.class
one/jfr/event/ObjectCount.class
one/jfr/event/Event.class
one/jfr/event/ContendedLock.class
one/jfr/event/EventAggregator.class
one/jfr/event/EventAggregator$Visitor.class
one/jfr/event/ExecutionSample.class
one/jfr/event/LiveObject.class
one/jfr/event/AllocationSample.class
one/jfr/event/CPULoad.class
one/profiler/AsyncProfilerLoader$ExecutionResult.class
one/profiler/AsyncProfilerLoader.class
one/profiler/AsyncProfiler.class
one/profiler/AsyncProfilerLoader$OSNotSupportedException.class
one/profiler/Events.class
one/profiler/Counter.class
one/profiler/AsyncProfilerMXBean.class
one/profiler/AsyncProfilerLoader$1.class

```

Reply:
@zhengziyi0117 Please take a look, if we could not copy codes, that is great.

Reply:
I think this is a great solution, but I still have the following problems: 
1. This package currently has some classes that I want to inherit that are not public, and some code may still need to be copied (I will try my best to solve this problem)
2. Some of the code in this package seems to be different from the current `async-profiler convert 3.0`, so the switch is a bit complicated.

I will raise an issue later to solve these problems.